apps/dashboard/src/components/contract-components/fetchPublishedContractsFromDeploy.ts

```diff
@@ -17,11 +24,36 @@ export async function fetchPublishedContractsFromDeploy(options: {
     throw new Error("No IPFS URI found in bytecode");
   }
 
-  const publishURIs = await getPublishedUriFromCompilerUri({
+  let publishURIs = await getPublishedUriFromCompilerUri({
     contract: getContractPublisher(client),
     compilerMetadataUri: contractUri,
   });
 
+  // Try fetching using contract name from compiler metadata for zksolc variants
+  // TODO: ContractPublisher should handle multiple metadata uri for a published version
+  if (publishURIs.length === 0 && (await isZkSyncChain(contract.chain))) {
+    const deployMetadata = await download({
+      uri: contractUri,
+      client,
+    })
+      // biome-ignore lint/suspicious/noExplicitAny:
+      .then((res) => res.json() as any)
+      .catch();
+    const contractId = Object.values(
+      deployMetadata?.source_metadata.settings?.compilationTarget,
+    );
+
+    await fetchPublishedContract({
+      client,
+      contractId: contractId[0] as string,
+      publisherAddress: THIRDWEB_DEPLOYER_ADDRESS,
+    })
+      .then((published) => {
+        publishURIs = [published.publishMetadataUri];
+      })
+      .catch();
```

Comment:
Prefer explicit try catch around the whole thing rather than 2 chained promises. In general better to inline everything rather than chain